src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1566,6 +1568,14 @@ OfferedIncompatibleQosStatus& DataWriterImpl::update_offered_incompatible_qos(
     return offered_incompatible_qos_status_;
 }
 
+LivelinessLostStatus& DataWriterImpl::update_liveliness_lost_status(
+        const fastrtps::LivelinessLostStatus& liveliness_lost_status)
+{
+    liveliness_lost_status_.total_count = liveliness_lost_status.total_count;
+    liveliness_lost_status_.total_count_change += liveliness_lost_status.total_count;
```

Comment:
I think this is rather brittle as it very much relies on the implementation details on WLP, but I do not know what to do about that